include/dr/shp/distributed_vector.hpp

```diff
@@ -175,6 +175,8 @@ struct distributed_vector {
 
   size_type size() const noexcept { return size_; }
 
+  size_type segment_size() const noexcept { return segment_size_; }
+
```

Comment:
Will this make it possible for the test to access private members instead of exporting public interfaces: https://stackoverflow.com/questions/46107103/accessing-private-members-with-googletest?

Reply:
Let's test public interface only. This method can be removed. Segment size can be checked by calling `segments` and checking its size.

Reply:
I'd prefer not to make the public-facing API too big, since calling `segments().size()` can already do this without introducing overhead.

Reply:
Done. Thanks. Please re-review this pr.